src/lock-fields-manager.ts

```diff
@@ -462,9 +492,14 @@ export class LockFieldsManager<U extends BaseMongoObject> {
 		if (existingEntityArray?.length > newEntityElement?.length) {
 			for (const newEntityElementArrayElement of newEntityElement) {
 				const elementAlreadyExists = _.some(ret, (existingElement) => {
-					return codeKeyNames?.every((key) => {
-						return existingElement[key] === _.get(newEntityElementArrayElement, key);
-					});
+					if (codeKeyNames) {
+						return this.elementsHaveSameReferences(
+							existingElement,
+							newEntityElementArrayElement,
+							codeKeyNames,
+						);
+					}
+					return _.isEqual(existingElement, newEntityElementArrayElement);
```

Comment:
a test is missing for this case

Reply:
as discussed, this part has been removed